utils.go

```diff
@@ -288,15 +288,15 @@ func (s *Service) detectLocation(name string) (location string, err error) {
 	defer func() {
 		err = s.formatError("detect_location", err, "")
 	}()
-
-	url := fmt.Sprintf("%s://%s.%s:%d", s.config.Protocol, name, s.config.Host, s.config.Port)
+	
+	url := fmt.Sprintf("%s://%s/%s", s.config.Protocol, "qingstor.com" , name )
 
 	r, err := s.client.Head(url)
 	if err != nil {
 		return
 	}
-	if r.StatusCode != http.StatusTemporaryRedirect {
-		err = fmt.Errorf("head status is %d instead of %d", r.StatusCode, http.StatusTemporaryRedirect)
+	if r.StatusCode != http.StatusMovedPermanently {
+		err = fmt.Errorf("head status is %d instead of %d", r.StatusCode, http.StatusNotFound)
```

Comment:
`StatusNotFound` should be `StatusMovedPermanently`.

Reply:
fix